### PR TITLE
Initial Kerberos Management

### PR DESCRIPTION
**Pull request by dmyerscough**
Initial kerberos management
- List principals/policies/
- Get principal/policy/privs
- Create/delete principal
- Create keytab

### CODE REVIEW

File: salt/modules/kerberos.py [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+    return False
+
+
+def __execute_kadmin(cmd):
+    '''
+    Execute kadmin commands
+    '''
+    ret = {}
+
+    auth_keytab = __opts__.get('auth_keytab', None)
```
Comment (thatch45) [inline salt/modules/kerberos.py]: We should be able to use config.get for these values, and they should likely be renamed to something like
krb:auth_keytab

**Comment by dmyerscough** [comment]
I will correct this at the weekend and create a new PR.

**Comment by gczuczy** [comment]
Please be careful if you add principal creation features. That can have a number of pitfalls, especially if one is using an LDAP backend, then you also have to support the -x dbargs for addprinc, otherwise the feature doesn't have any use in such a scenario.

Later will do some testing, atm my hands are full at work.

**Comment by dmyerscough** [comment]
@gczuczy Please can you give me an example of creating a principal using LDAP and I can look at getting that feature added.

**Comment by gczuczy** [comment]
Sure. Please first read this:
http://web.mit.edu/kerberos/krb5-devel/doc/admin/conf_ldap.html
Then please read this thread on actually how aliases and linked principals work:
http://web.mit.edu/kerberos/krb5-devel/doc/admin/conf_ldap.html

So, from now. Let's say you have a user foo, for dn= cn=Foo,ou=users,dc=example,dc=com, and adding a principal for this user would be:
addprinc -x dn="cn=foo,ou=users,dc=example,dc=com" foo@REALM

This will put all the kerberos stuff to the user's DIT. The above mail thread explains how to use the krbCannonicalName + multivalued krbPrincipalName to have aliases, and what's the different when you do the above addprinc, just with linkdn= as a dbarg.

**Pull request merged**